### PR TITLE
Afoxman/build tweaks

### PR DESCRIPTION
**Pull request by afoxman**
Move away from using a toolchain file on Windows (setting the compiler explicitly isn't a great idea). Push settings into the CI loops and the CMake logic instead.

### CODE REVIEW

**Comment by afoxman** [comment]
There's a lot of white-space changes in here, due to the recent changes I made to enable VSCode settings for the project. Specifically, the trim trailing whitespace. These will stop showing up as we touch the various files. Sorry for the noise.

File: tools/cmake/liblet.cmake [hunk 1]
```diff
@@ -45,6 +45,13 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   set(CMAKE_CXX_ARCHIVE_FINISH "<CMAKE_RANLIB> -no_warning_for_no_symbols -c <TARGET>")
 endif()
 
+if(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
+  set(WIN_TARGET_ARCH $ENV{VSCMD_ARG_TGT_ARCH})
```
Comment (afoxman) [inline tools/cmake/liblet.cmake:49]: @vmoroz In the other draft review, you were asking about this variable and how it was set. VSCMD_ARG_TGT_ARCH is set by the Visual Studio command shell (vcvarsall.bat and its wrappers). When it is missing, I fail with a message (ln 51) that asks the user if they are in a Visual Studio shell, to help them realize this is a requirement.

Comment (vmoroz) [reply tools/cmake/liblet.cmake:49]: Makes sense. Thank you!

File: .ado/vcenv.bat [hunk 2]
```diff
@@ -0,0 +1,3 @@
+CALL 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvarsall.bat' %BUILDPLATFORM% %BUILDWINDOWSSDK%
```
[the remark this answers is not in the document]
Comment (afoxman) [reply .ado/vcenv.bat]: For the ADO CI machines, they publish explicit locations for everything that is pre-installed. They don't set environment variables (that I know of) or have any kind of config / registration system that lets you find them by name.

[the remark this answers is not in the document]
Comment (afoxman) [reply .ado/vcenv.bat]: https://github.com/actions/virtual-environments/blob/master/images/win/Windows2019-Readme.md

File: .ado/templates/windows-build-job.yml [hunk 3]
```diff
@@ -59,15 +59,14 @@ jobs:
...
-        filename: 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvarsall.bat'
-        arguments: $(BuildPlatform) $(BuildWindowsSDK)
+        filename: vcenv.bat
         modifyEnvironment: true
 
     - task: CMake@1
       displayName: Generate Ninja scripts
       inputs:
         workingDirectory: $(Build.BinariesDirectory)/$(MsoPlatform)/$(BuildConfiguration)/$(BuildPlatform)
-        cmakeArgs: -G Ninja -DMSO_LIBLET_PLATFORM=$(MsoPlatform) -DCMAKE_BUILD_TYPE=$(BuildConfiguration) -DCMAKE_TOOLCHAIN_FILE=$(Build.SourcesDirectory)\tools\CMake\toolchain.windows.cmake $(Build.SourcesDirectory)
```
[the remark this answers is not in the document]
Comment (afoxman) [reply .ado/templates/windows-build-job.yml:70]: Setting CMAKE_C_COMPILER and CMAKE_CXX_COMPILER is discouraged, and that's what I was using it for. I've also noticed that CMake does strange things when a toolchain file is in play, like regenerating itself when it shouldn't. It also seems to foul up the integrated CMake in VSCode and VS.

File: CMakeLists.txt [hunk 4]
```diff
@@ -5,10 +5,32 @@
...
+# When our project is the root project, we are building stand-alone and can
+# can safely manipulate the warning level. Turn on all warnings, and treat
+# them as errors. This ensures that when we are imported, we will build
+# correctly in even the most restrictive projects.
+if(${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_CURRENT_SOURCE_DIR})
+  if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+    string(REGEX REPLACE "[/-][wW]3" "" CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /permissive- /W4 /WX")
+  elseif(CMAKE_CXX_COMPILER_ID IN_LIST "AppleClang;Clang")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pedantic -Wall -Wextra -Werror")
```
[the remark this answers is not in the document]
Comment (afoxman) [reply CMakeLists.txt]: I removed this (going to use target options instead and a new cmake policy in 3.15).

**Pull request merged**